### DISCUSSION TITLE
Whether stack trace is necessary, when GROUP share level is not used ？

### DISCUSSION THREAD

**Discussion by merrily01** [apache/kyuubi#4512] [General]
Under version 1.7，

When I start the beeline connection, 

The stack trace always appears in the client log, as follows:

<img width="1425" alt="image" src="https://user-images.githubusercontent.com/15688792/224999635-a36afd3c-d485-411e-8a1f-9b828a00430d.png">

I found that it is because after the pr of  https://github.com/apache/kyuubi/pull/3897 is merged, When we create a new EngineRef, no matter which engine share level is used, getGroupNames will always be called. 

Therefore, when the user does not use the GROUP mode, is it necessary to prompt the stack message above ？

**Comment by merrily01** [comment]
Kindly cc @pan3793

**Reply by pan3793** [reply]
Thanks for reporting, looks like it's fine to change the stack trace to debug, and just info "There is no group for <user>, use the client user name as group directly." Would you like to send a PR to fix it?

More thought, caching the `HadoopGroupProvider` result or lazy evaluating the group name may be another optimization direction.

**Reply by merrily01** [reply]
Yeah, i will work on this.
Thanks for your suggestion :)

**Reply by pan3793** [reply]
link a previous discussion about GroupProvider in case you may be interested.
https://github.com/apache/kyuubi/discussions/4478

**Reply by merrily01** [reply]
Got it！ Thanks again

**Comment by yaooqinn** [comment]
let's fix this by moving sessionMgr.groupProvider inside EngineRef.appUser
```scala
   private[kyuubi] val appUser: String = shareLevel match {
     case SERVER => Utils.currentUser
-    case GROUP => primaryGroup
+    case GROUP => sessionMgr.groupProvider.primaryGroup(user, conf.getAll.asJava)
     case _ => user
   }
```

**Reply by merrily01** [reply]
Okay, 

I was just thinking that this stack trace looks like a log converted after an internal call catches an exception, which is not easy to mask directly.

thank you for reminding.

**Reply by yaooqinn** [reply]
Would you mind to report this bug to the GitHub issue page for better issue-tracking

**Reply by merrily01** [reply]
Done, i have reported a issue: https://github.com/apache/kyuubi/issues/4526 , and made a pr : https://github.com/apache/kyuubi/pull/4527

**Reply by yaooqinn** [reply]
thank you

**Comment by merrily01** [comment]
FYI : 

Releated issue :  https://github.com/apache/kyuubi/issues/4526

Solved by pr :  https://github.com/apache/kyuubi/pull/4527

**Reply by yaooqinn** [reply]
thank you @merrily01
